flagd/pkg/service/flag-evaluation/flag_evaluator.go

```diff
@@ -172,12 +181,20 @@ func (s *OldFlagEvaluationService) ResolveBoolean(
 	sCtx, span := s.flagEvalTracer.Start(ctx, "resolveBoolean", trace.WithSpanKind(trace.SpanKindServer))
 	defer span.End()
 	res := connect.NewResponse(&schemaV1.ResolveBooleanResponse{})
+	evalCtx := map[string]any{}
+	if e := req.Msg.GetContext(); e != nil {
+		evalCtx = e.AsMap()
+	}
+	for k, v := range s.contextValues {
+		evalCtx[k] = v
+	}
```

Comment:
This code block exists multiple times; changing how we populate the context means we must touch several parts of the code. We should extract this into our method for easier maintenance and effortless changes.

Furthermore, I think it is worth moving this code block into the resolve method and handling it once before resolving the flag.